### PR TITLE
Grid: taking an actor's scale into account when evaluating actor scene constraint

### PR DESCRIPTION
**Pull request by Juanpe**
Now we use the actor's ```rect```, instead of actor's ```size```, because the ```rect``` is already adjusted for its scale.

### CODE REVIEW

**Comment by JohnSundell** [comment]
@Juanpe Beautiful! 👌 Thanks for this!

Now we just need to add a test to get this merged into master 👍 There's already a test in `ActorTests` called `testConstrainingToScene`, but I suggest adding an additional one that makes sure that the constraint evaluation takes scale into account. Let me know if you want any help or pointers on how to implement the test  🙂

**Comment by Juanpe** [comment]
I've added an additional check inside the method called ```testConstrainingToScene```. Is enough or should be better create another test method only for this.

File: Tests/ImagineEngineTests/ActorTests.swift [hunk 1]
```diff
@@ -175,6 +175,15 @@ final class ActorTests: XCTestCase {
...
         XCTAssertEqual(actor.position.y, 450)
+        
+        // Set actor scale
+        actor.scale = 0.5
+
+        actor.position.x = 500
+        XCTAssertEqual(actor.position.x, 475)
+        
+        actor.position.y = 600
+        XCTAssertEqual(actor.position.x, 475)
```
Comment (JohnSundell) [inline Tests/ImagineEngineTests/ActorTests.swift]: This should be `actor.position.y` 😉

**Comment by JohnSundell** [comment]
That's awesome @Juanpe! Just one little mistake in the test, but once that's fixed I'll merge this in 👍

File: Tests/ImagineEngineTests/ActorTests.swift (hunk 1 above)
Comment (Juanpe) [reply Tests/ImagineEngineTests/ActorTests.swift]: you're right 😅 the rushes are not good

**Comment by Juanpe** [comment]
Thanks! 🎉

**Comment by JohnSundell** [comment]
It works perfectly @Juanpe, thanks again for contributing 😄
fix

**Pull request merged**